File: Code-Review-Mcp/test-fixtures/boyscout-actions/dotnet/BoyscoutSample.cs
namespace BoyscoutActionsFixture;

public class BoyscoutSample
{
    private void UnusedHelper() => Console.WriteLine("never called");

    public string ReadFirst(List<string?> items)
    {
        return items.FirstOrDefault()!.Length.ToString();
    }

    public int PublishOrder(string orderId, List<(int Qty, decimal Price)> items)
    {
        var total = 0m;
        var valid = true;

        foreach (var item in items)
        {
            if (item.Qty <= 0)
            {
                valid = false;
                break;
            }
            if (item.Price < 0)
            {
                valid = false;
            }
        }
        if (!valid)
        {
            throw new InvalidOperationException("invalid");
        }

        foreach (var item in items)
        {
            var line = item.Qty * item.Price;
            if (orderId.StartsWith("VIP"))
            {
                line *= 0.9m;
            }
            else if (orderId.StartsWith("BULK"))
            {
                line *= 0.85m;
            }
            if (line > 500)
            {
                line *= 0.95m;
            }
            else if (line > 100)
            {
                line *= 0.98m;
            }
            total += line;
        }

        return (int)total;
    }
}

File: Codebase.Analyzer/test-fixtures/boyscout-actions/angular/boyscout-sample.ts
export class BoyscoutSample {
  private unusedHelper(): void {
    console.log("never called");
  }

  readDom(): string {
    const el = document.getElementById("panel");
    return el!.innerText;
  }

  public publishOrder(order: { id: string }, items: { qty: number; price: number }[]): number {
    let total = 0;
    let valid = true;

    for (const item of items) {
      if (item.qty <= 0) {
        valid = false;
        break;
      }
      if (item.price < 0) {
        valid = false;
      }
    }
    if (!valid) {
      throw new Error("invalid");
    }

    for (const item of items) {
      let line = item.qty * item.price;
      if (order.id.startsWith("VIP")) {
        line = line * 0.9;
      } else if (order.id.startsWith("BULK")) {
        line = line * 0.85;
      }
      if (line > 500) {
        line = line * 0.95;
      } else if (line > 100) {
        line = line * 0.98;
      }
      total += line;
    }

    return total;
  }
}
